src/plan/global.rs

```diff
@@ -621,6 +732,18 @@ impl<VM: VMBinding> CommonPlan<VM> {
         self.base.release(tls, full_heap)
     }
 
+    pub fn clear_side_log_bits(&self) {
+        self.immortal.clear_side_log_bits();
+        self.los.clear_side_log_bits();
+        self.base.clear_side_log_bits();
+    }
+
+    pub fn set_side_log_bits(&self) {
+        self.immortal.set_side_log_bits();
+        self.los.set_side_log_bits();
+        self.base.set_side_log_bits();
+    }
```

Comment:
It looks like neither `clear_side_log_bits` nor `set_side_log_bits` are methods of any traits, but methods of specific spaces that support log bits.  We may extract a trait and make it a trait method.

```rust
pub trait SupportsLogBits: Space {
    fn clear_side_log_bits(&self);
    fn set_side_log_bits(&self);
}
```

Then when needed, we can simply iterate through all spaces and call that method.

```rust
fn prepare(&self) {
            self.for_each_space(|space| {
                if let Some(s) = space.downcast::<SupportsUnlogBits>() {
                    s.clear_side_log_bits();
                }
            });
}
```

Reply:
I added those methods to `trait Space`. All the spaces should allow plans to manipulate log bits on them. However, for some policies, we are not using those methods for the current plans. I leave them as `unimplemented!()`.